### ISSUE TITLE
Windows command interop in WSL does not work, reports not found for all commands

### ISSUE THREAD

**Issue by Neutrino-Sunset** [microsoft/WSL#6243]
# Environment

```
Windows build number: Microsoft Windows [Version 10.0.18363.1198]
Your Distribution version: [On Debian or Ubuntu run `lsb_release -r` in WSL]
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020
```

# Steps to reproduce

Installed Docker Desktop and the patch to update WSL to 2 by following these instructions [WSL Install Instructions](https://docs.microsoft.com/en-us/windows/wsl/install-win10) but then installing Docker Desktop instead of a standalone distro.

Then:
```
    C:\Users\User>wsl -l -v
      NAME                   STATE           VERSION
    * docker-desktop         Running         2
      docker-desktop-data    Running         2
```

Note I have no additional distribs installed other than what Docker installs itself.

Run wsl to start the default distrib which `wsl -l -v` indicated was `docker-desktop`. Then from inside that distrib execute a simple Windows command using interop as described in the documentation [Run Windows Tools From Linux](https://docs.microsoft.com/en-us/windows/wsl/interop#run-windows-tools-from-linux)
```
C:\Users\User>wsl
DESKTOP-SDE0C3N:/tmp/docker-desktop-root/mnt/host/c/Users/Martin# cmd.exe /C dir
-sh: cmd.exe: not found
```

That's a 15 minute process! Don't you have a half reasonable method of getting some logs? 

#  Expected behavior

I'm expecting cmd.exe to run.

# Actual behavior

It fails with `Not Found`.

**Comment by DHowett** [comment]
You will need to install an actual distribution like Ubuntu or Debian. The docker distributions are intended to support the docker product and are not for general-purpose use.

**Comment by Neutrino-Sunset** [comment]
Fair enough. But in that case surely either `wsl -l` shouldn't list the Docker distros as distros at all, or the [WSL interop documentation](https://docs.microsoft.com/en-us/windows/wsl/install-win10) should be clear that the features it describes applies to some distros and not others, and WSL should ideally provide some means of identifying "real" distros from ones in which key features will not work as described.

**Comment by therealkenc** [comment]
There is open #6024 which is _nearly_ sufficient as an open landing zone for all things "docker is my default", but this submission could be coerced into a standalone feature-request "do not list docker distros in wsl -l -v" or somesuch.

**Comment by HeyITGuyFixIt** [comment]
Not sure if this is the same issue, but WSL interop doesn't appear to work when running in a root shell, like in the example you gave above. I am running Ubuntu-20.04 distro with WSL2. For example, if I am in a root shell and run `notepad.exe` I get `notepad.exe: command not found`. So then I run `/mnt/c/windows/system32/notepad.exe` and I get no output, and nothing happens. I originally tried it with `quser` and I got the same results. When I am not in a root shell, and just in a regular user shell, the WSL interop stuff works fine.
